playbooks/robusta_playbooks/playbooks_utils.py

```diff
@@ -0,0 +1,175 @@
+    internal_ip = next(
+        addr.address
+        for addr in node.status.addresses
+        if addr.type == "InternalIP"
+    )
+    return internal_ip
+
+
+def __prepare_promql_query(provided_labels: Dict[Any, Any], promql_query_template: str) -> str:
+    labels = defaultdict(lambda: "<missing>")
+    labels.update(provided_labels)
+    if '$node_internal_ip' in promql_query_template:
+        # TODO: do we already have alert.Node here?
+        node_name = labels['node']
+        node: Node = Node.readNode(node_name).obj
+        if not node:
+            logging.warning(
+                f"Node {node_name} not found for"
+            )
+            raise AttributeError(f'Cannot get internal ip for node: {node_name}')
+        node_internal_ip = __get_node_internal_ip_from_node(node)
+        labels['node_internal_ip'] = node_internal_ip
+    template = Template(promql_query_template)
+    promql_query = template.safe_substitute(labels)
+    return promql_query
+
+
+def create_chart_from_prometheus_query(
+        prometheus_base_url: str,
+        promql_query: str,
+        starts_at: datetime,
+        include_x_axis: bool,
+        graph_duration_minutes: int,
+        chart_title: Optional[str] = None,
+        values_format: Optional[ChartValuesFormat] = None
+):
+    if not prometheus_base_url:
+        prometheus_base_url = PrometheusDiscovery.find_prometheus_url()
+    prom = PrometheusConnect(url=prometheus_base_url, disable_ssl=True)
+    end_time = datetime.now(tz=starts_at.tzinfo)
+    alert_duration = end_time - starts_at
+    graph_duration = max(alert_duration, timedelta(minutes=graph_duration_minutes))
+    start_time = end_time - graph_duration
+    resolution = 250  # 250 is used in Prometheus web client in /graph and looks good
+    increment = max(graph_duration.total_seconds() / resolution, 1.0)
+    result = prom.custom_query_range(
+        promql_query,
+        start_time,
+        end_time,
+        increment,
+        {"timeout": PROMETHEUS_REQUEST_TIMEOUT_SECONDS},
+    )
+
+    chart = pygal.XY(
+        show_dots=True,
+        style=ChosenStyle,
+        truncate_legend=15,
+        include_x_axis=include_x_axis,
+        width=1280,
+        height=720
+    )
+
+    chart.x_label_rotation = 35
+    chart.truncate_label = -1
+    chart.x_value_formatter = lambda timestamp: datetime.fromtimestamp(
+        timestamp
+    ).strftime("%I:%M:%S %p on %d, %b")
+
+    value_formatters = {
+        ChartValuesFormat.Plain: lambda val: str(val),
+        ChartValuesFormat.Bytes: lambda val: humanize.naturalsize(val, binary=True),
+        ChartValuesFormat.Percentage: lambda val: f'{(100*val):.1f}'
+    }
+    chart_values_format = values_format if values_format else ChartValuesFormat.Plain
+    chart.value_formatter = value_formatters[chart_values_format]
+
+    if chart_title:
+        chart.title = f'{chart_title} starting {humanize.naturaldelta(timedelta(minutes=graph_duration_minutes))} ago'
```

Comment:
I'm not crazy about that format.
Can we ask the team what they think the best format is, and get data driven decision?